src/drivers/imu/analog_devices/adis16507/ADIS16507.cpp

```diff
@@ -66,20 +67,22 @@ int ADIS16507::init()
 	int ret = SPI::init();
 
 	if (ret != PX4_OK) {
-		DEVICE_DEBUG("SPI::init failed (%i)", ret);
+		PX4_ERR("SPI::init failed (%i)", ret);
 		return ret;
 	}
 
-	return Reset() ? 0 : -1;
+	_state = STATE::RESET;
+	ScheduleDelayed(1_ms);
+
+	return PX4_OK;
 }
 
-bool ADIS16507::Reset()
+void ADIS16507::Reset()
 {
 	_state = STATE::RESET;
 	DataReadyInterruptDisable();
 	ScheduleClear();
-	ScheduleNow();
-	return true;
+	ScheduleDelayed(1_ms);
```

Comment:
Why 1 ms? If the delay matters add a comment, otherwise don't sprinkle in magic timing.